### PR TITLE
Create bogus-error-#39161.rs

### LINKED ISSUE #39161: Bogus error when using `..` in struct expr with all fields present, in a const ("error[E0016]: blocks in constants are limited to items and tail expressions")
```rust
struct X { a: i32, b: i32 }

#[allow(unused_variables)]
fn main() {
    const DX: X = X { a: 0, b: 0 };
    const X1: X = X { a: 1, ..DX };  // ok
    let   x2    = X { a: 1, b: 2, ..DX };  // ok
    const X3: X = X { a: 1, b: 2, ..DX };  // error[E0016]
}
```

* [ ] It's weird that `X1` and `x2` are allowed, but not `X3`.
* [ ] Even if that's intentional, the error message seems off.

### PR DESCRIPTION
**Pull request by YesSeri**
closes #39161 

It is still working.

### CODE REVIEW

File: src/test/ui/bogus-error-#39161.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// run-pass
```
Comment (JohnTitor) [inline src/test/ui/bogus-error-#39161.rs:1]: I think this could be `// check-pass`?

**Review by JohnTitor — COMMENTED** [review]
So, I think this test would be better:
```rust
pub struct X {
    pub a: i32,
    pub b: i32,
}

fn main() {
    const DX: X = X { a: 0, b: 0 };
    const _X1: X = X { a: 1, ..DX };
    let _x2 = X { a: 1, b: 2, ..DX };
    const _X3: X = X { a: 1, b: 2, ..DX };
}
```

This doesn't need allow attribute and emit warnings.

**Comment by JohnTitor** [comment]
Also, could you place this test in `src/test/ui/consts` and rename to `issue-39161-bogus-error.rs`?

**Comment by Muirrum** [comment]
@YesSeri this is a triage bump.

**Comment by nikomatsakis** [comment]
I'm going to go ahead and close this PR, as it's over 3 months old. @YesSeri please feel free to re-open if you get a chance to address @JohnTitor's feedback. ❤️

**Pull request closed without merging**